Tests/SentryTests/Integrations/Performance/Network/SentryNetworkTrackerTests.swift

```diff
@@ -559,6 +509,60 @@ class SentryNetworkTrackerTests: XCTestCase {
         //Test if it has observers. Nil means no observers
         XCTAssertNil(task.observationInfo)
     }
+
+    func testBaggageHeader() {
```

Comment:
I wonder if we still need the test-server, and those 2 tests that do an actual request to the test server. All we're really testing is if it headers are added, which we can test way easier like I added here.

Reply:
Because the way we swizzle  NSSessionTask, we can't guarantee that the header injection is working on every iOS release, we still need to do a integration test.

Reply:
For which iOS versions is this needed? Is this something we can clean up with 8.0.0?

Reply:
Ah no because now we manually call `urlSessionTaskResume` and thus kicking off the test, but that skips the swizzling. Still feels a bit heavy handed, this test server just to test sending 2 headers that are not needed for Sentry itself to work. Oh well.

Reply:
Using swizzling has a risk of breaking actual HTTP requests. We still use swizzling for HTTP requests. I would rather keep one integration test in combination with the test server. We don't have to test all the functionality with the test server.